Lib/ast.py

```diff
@@ -213,6 +226,47 @@ def get_docstring(node, clean=True):
     return text
 
 
+def get_source_segment(source, node, *, padded=False, coding='utf-8'):
+    """Get source code segment of the *source* that generated *node*.
+
+    If some location information (`lineno`, `end_lineno`, `col_offset`,
+    or `end_col_offset`) is missing, return None. The default *coding*
+    is UTF-8 (the one used by the Python parser).
+
+    If *padded* is `True`, the first line of a multi-line statement will
+    be padded with spaces to match its original position.
+    """
+    if not hasattr(node, 'lineno'):
+        return None
+    lineno = node.lineno - 1
+    if not hasattr(node, 'end_lineno'):
+        return None
+    end_lineno = node.end_lineno - 1
+    if not hasattr(node, 'col_offset'):
+        return None
+    col_offset = node.col_offset
+    if not hasattr(node, 'end_col_offset'):
+        return None
+    end_col_offset = node.end_col_offset
+
+    if end_lineno == lineno:
+        source_line = source.splitlines()[lineno]
+        return source_line.encode()[col_offset:end_col_offset].decode(coding)
```

Comment:
I think *coding* should not be used here. Always use UTF-8.

Reply:
You are totally right.